### ISSUE TITLE
rust bindings "node_kind_is_visible" is inaccurate

### ISSUE THREAD

**Issue by alanz** [tree-sitter/tree-sitter#997]
[node_kind_is_visible](https://github.com/tree-sitter/tree-sitter/blob/f3910c649e5c90438e6a86a7d29e4554ca7aeb12/lib/binding_rust/lib.rs#L235-L239) is defined as

```rust
pub fn node_kind_is_visible(&self, id: u16) -> bool {
        unsafe {
            ffi::ts_language_symbol_type(self.0, id) <= ffi::TSSymbolType_TSSymbolTypeAnonymous
        }
```

Where [ts_language_symbol_type](https://github.com/tree-sitter/tree-sitter/blob/48584c7cadb66b8d08b2dfed704cfab748a02d41/lib/src/language.c#L93-L105) is defined as

```c
TSSymbolType ts_language_symbol_type(
  const TSLanguage *self,
  TSSymbol symbol
) {
  TSSymbolMetadata metadata = ts_language_symbol_metadata(self, symbol);
  if (metadata.named) {
    return TSSymbolTypeRegular;
  } else if (metadata.visible) {
    return TSSymbolTypeAnonymous;
  } else {
    return TSSymbolTypeAuxiliary;
  }
}
```

This means that a symbol having [metadata](https://raw.githubusercontent.com/AbstractMachinesLab/tree-sitter-erlang/main/src/parser.c) as

```
  [sym__module] = {
    .visible = false,
    .named = true,
  },
```

is erroneously reported as being visible.

This function should call `ts_language_symbol_metadata` and return the `visible` field value directly.

**Comment by patrickt** [comment]
This looks reasonable to me. `ts_language_symbol_metadata` doesn’t have a Rust bridge, so we’ll either have to bridge that or bridge some helper `ts_language_symbol_is_visible` function. I’ll defer to @maxbrunsfeld on that, but can bang this out either way.

**Comment by maxbrunsfeld** [comment]
Oh, that `ts_language_symbol_type` is actually used similarly in the WASM binding. I think we should change its definition so that it only return` TSSymbolTypeRegular` if the `TSSymbolMetadata` is both `visible` and `named`.

**Comment by maxbrunsfeld** [comment]
Ok, this is fixed via #998.
